### PR TITLE
Improved scalafiddle integration

### PR DESCRIPTION
**Pull request by andreaTP**
Here we adopt an higher-level integration with scalafiddle leveraging: https://github.com/scalafiddle/scalafiddle-core/tree/master/integrations

This includes (mostly thanks to @ochrons ):
 - better and more readable tests
 - consistent offline code show (I recall we have an issue tracking this ...)
 - nicer UX
 - much less hacky interface

Note:
 - this completely break compatibility (since we are in `0.X` I guess it's fine)

cc. @ochrons @2m @patriknw @ktoso

### CODE REVIEW

**Comment by andreaTP** [comment]
I will check consistency and update the Akka docs in the next days possibly (certainly before finalizing this PR)

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.3.6-SNAPSHOT"
+version in ThisBuild := "0.4.0-SNAPSHOT"
```
Comment (patriknw) [inline version.sbt]: what is the reason for the "big" bump?

Comment (andreaTP) [reply version.sbt]: the fiddle integration breaks the API

**Comment by andreaTP** [comment]
I tested it publishing everything with akka-docs locally from the other PR branches, not sure how to publish everything ...

File: version.sbt (hunk 1 above)
Comment (andreaTP) [reply version.sbt]: e.g. a `0.3` fiddle directive will no work with `0.4`

Comment (patriknw) [reply version.sbt]: ok, then the bump makes sense

**Comment by ktoso** [comment]
LGTM, thanks

**Comment by ktoso** [comment]
Give us time... soon.

**Pull request merged**